File: src/test/java/br/com/zumbolovsky/fateapp/ContainersInitializer.java
package br.com.zumbolovsky.fateapp;

import org.springframework.context.ApplicationContextInitializer;
import org.springframework.context.ConfigurableApplicationContext;

public class ContainersInitializer implements ApplicationContextInitializer<ConfigurableApplicationContext> {

    @Override
    public void initialize(ConfigurableApplicationContext configurableApplicationContext) {

        // Initialize and start test containers
        new PostgreSQLSetup();
        new MongoDBSetup();
        new RedisSetup();
    }
}

File: src/test/kotlin/br/com/zumbolovsky/fateapp/ContainersInitializer.kt
package br.com.zumbolovsky.fateapp

import org.springframework.context.ApplicationContextInitializer
import org.springframework.context.ConfigurableApplicationContext

internal class ContainersInitializer : ApplicationContextInitializer<ConfigurableApplicationContext> {
    override fun initialize(configurableApplicationContext: ConfigurableApplicationContext) {

        // Initialize and start test containers
        PostgreSQLSetup()
        MongoDBSetup()
    }
}